src/ripple/ledger/impl/ReadView.cpp

```diff
@@ -81,18 +92,21 @@ class Rules::Impl
 
 //------------------------------------------------------------------------------
 
-Rules::Rules (DigestAwareReadView const& ledger)
-    : impl_(std::make_shared<Impl>(ledger))
+Rules::Rules(
+    DigestAwareReadView const& ledger,
+        std::unordered_set<uint256, beast::uhash<>> const& presets)
+    : impl_(std::make_shared<Impl>(ledger, presets))
+{
+}
+
+Rules::Rules(std::unordered_set<uint256, beast::uhash<>> const& presets)
+    : impl_(std::make_shared<Impl>(presets))
 {
 }
 
 bool
-Rules::enabled (uint256 const& id,
-    std::unordered_set<uint256,
-        beast::uhash<>> const& presets) const
+Rules::enabled (uint256 const& id) const
 {
-    if (presets.count(id) > 0)
-        return true;
```

Comment:
`Rules::enabled (uint256 const& id) const`

The new first line is `if(!impl_)`. I figure since there's no default ctor now, `impl_` should never be null, so we could `assert(impl_)`.